api/src/main/java/org/pmiops/workbench/db/dao/UserRecentWorkspaceDao.java

```diff
@@ -0,0 +1,10 @@
+package org.pmiops.workbench.db.dao;
+
+import org.pmiops.workbench.db.model.UserRecentWorkspace;
+import org.springframework.data.repository.CrudRepository;
+
+import java.util.List;
+
+public interface UserRecentWorkspaceDao extends CrudRepository<UserRecentWorkspace, Long> {
+    List<UserRecentWorkspace> findTopByUserIdOrderByLastUpdateTime(long userId);
```

Comment:
This method doesn't appear to take in a max count, so it could be thousands of workspaces if the user was running a script potentially.

I think you want to order by update time descending maybe?